### PR TITLE
Implement arrow functions fn() =>

### PR DESCRIPTION
**Pull request by kylekatarnls**
# Arrow functions
https://www.php.net/manual/en/migration74.new-features.php#migration74.new-features.core.arrow-functions

```php
function doubleAll(array $numbers) {
    return array_map(fn($number) => $number * 2, $numbers);
}
```

Fix https://github.com/phpmd/phpmd/issues/726

### CODE REVIEW

**Comment by kylekatarnls** [comment]
This requires #422 first to be merged so it can be checked with PHP 7.4 on Travis-CI.

**Comment by tvbeek** [comment]
@kylekatarnls can you check the failing test before I review this?

**Comment by kylekatarnls** [comment]
Yes, it should not be too harmful, it's the test that check arrow function would throw an error with PHP 7.3 parser. It's kind of a nice to have but we could live without it. By default PDepend is loaded with maximum version level, only a specific usage by requiring the specific PHPXParser class would actually run this code, and yet many features do not have error thrown when used in a lower level than the one they would request.

File: src/main/php/PDepend/Source/Language/PHP/PHPTokenizerInternal.php [hunk 1]
```diff
@@ -181,137 +188,138 @@ class PHPTokenizerInternal implements Tokenizer
...
-        T_IS_GREATER_OR_EQUAL       =>  Tokens::T_IS_GREATER_OR_EQUAL,
-        T_IS_SMALLER_OR_EQUAL       =>  Tokens::T_IS_SMALLER_OR_EQUAL,
-        //T_PAAMAYIM_NEKUDOTAYIM      =>  Tokens::T_DOUBLE_COLON,
-        T_ENCAPSED_AND_WHITESPACE   =>  Tokens::T_ENCAPSED_AND_WHITESPACE,
-        T_CONSTANT_ENCAPSED_STRING  =>  Tokens::T_CONSTANT_ENCAPSED_STRING,
-        T_YIELD                     =>  Tokens::T_YIELD,
-        T_FINALLY                   =>  Tokens::T_FINALLY,
-        T_COALESCE                  =>  Tokens::T_COALESCE,
-        //T_DOLLAR_OPEN_CURLY_BRACES  =>  Tokens::T_CURLY_BRACE_OPEN,
+        T_AS                        => Tokens::T_AS,
```
Comment (tvbeek) [inline src/main/php/PDepend/Source/Language/PHP/PHPTokenizerInternal.php]: This list of changes wasn't really needed but in this smaller PR I didn't see a problem in cleaning this.

Comment (kylekatarnls) [reply src/main/php/PDepend/Source/Language/PHP/PHPTokenizerInternal.php]: It's T_FN + indent fix, there is no list change.

Comment (tvbeek) [reply src/main/php/PDepend/Source/Language/PHP/PHPTokenizerInternal.php]: Sorry I wasn't very clear. The indent made it looks like a big list 😃

**Pull request merged**